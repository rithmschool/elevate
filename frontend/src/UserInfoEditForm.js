import React from 'react';
import { Col, Button, Form, Label, Input, Row} from 'reactstrap';
import './UserInfoEditForm.css';
import moment from 'moment';



/** Update user basic info and salary */

class UserInfoEditForm extends React.Component {
  
  constructor(props){
    super(props);
    this.state = {
      first_name: this.props.first_name,
      last_name: this.props.last_name,
      email: this.props.email,
      current_company: this.props.current_company || '',
      salary: this.props.salary, 
      bonus: this.props.bonus,
      equity: this.props.equity,
      hire_date: moment(this.props.hire_date).format('YYYY-MM-DD'),
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
 // sync state values with inputs values
  handleChange(evt){
		this.setState({ [evt.target.name]: evt.target.value });
  }
  
  handleSubmit(evt){
    evt.preventDefault()
    //update user basic info
    const updatedUser = {
      first_name: this.state.first_name,
      last_name: this.state.last_name,
      email: this.state.email,
      current_company: this.state.current_company,
      hire_date: this.state.hire_date
    }
    this.props.handleUserUpdate(updatedUser, this.props.userId);

    //update salary
    const salary = {
      user_id: this.props.userId,
      salary: this.state.salary,
      bonus: this.state.bonus,
      equity: this.state.equity
    };
    this.props.handleSalaryUpdate(salary, updatedUser.current_company);
  }
  
  render() {
    const isEdit = this.props.isEdit
    return (
      <div className="EditPUserForm container border rounded"
        style={{backgroundColor:'#F4F6F8'}}>
        <Row>
          <Col md={6}>
            <h3 className="text-info">User information</h3>
          </Col>
          <Col align="right" md={6} sm={{ size: 8, offset: 0}}>
          <i className="fas fa-edit text-info"
            onClick={this.props.toggleEdit} 
            style={{cursor: 'pointer', fontSize: '1.1em', marginTop: '1em' }}></i>
          </Col>
        </Row>

        <hr></hr>        
      <Form onSubmit={this.handleSubmit}> 
      <h5>Basic info</h5><br></br>
        <Row form>
          <Col md={4}>                  
            <Label className="form-group has-float-label">
              <Input onChange={this.handleChange}
                    value={this.state.first_name}
                    type="text"
                    name="first_name"
                    id="EditUser-first_name"
                    disabled={!isEdit}  />
              <span>First name</span>
            </Label>
            </Col>
          <Col md={4}>
            <Label className="form-group has-float-label">
            <Input onChange={this.handleChange}
                    value={this.state.last_name}
                    type="text" name="last_name"
                    id="EditUser-last_name" 
                    disabled={!isEdit}/>
            <span>Last name</span>
            </Label>
          </Col>
          <Col md={4}>
            <Label className="form-group has-float-label">
            <Input onChange={this.handleChange}
                    value={this.state.email}
                    type="email" name="email"
                    id="EditUser-email"
                    disabled={!isEdit}/>
            <span>Email</span>
            </Label>
          </Col>
        </Row>
        <hr></hr>
        <h5>Company info</h5><br></br>

        <Row form> 
        <Col md={6}>
            <Label className="form-group has-float-label">  
            <Input onChange={this.handleChange}
                    value={this.state.current_company}
                    type="text" name="current_company"
                    id="EditUser-current_company" 
                    disabled={!isEdit}/>
            <span>Current company</span>
            </Label>
          </Col> 
          <Col md={6}>
            <Label className="form-group has-float-label"> 
            <Input onChange={this.handleChange}
                    value={this.state.hire_date}
                    type="date" name="hire_date"
                    id="EditUser-hire_date"
                    disabled={!isEdit}
                    />
            <span>Hire date</span>
            </Label>
          </Col>
          </Row>
          <Row form> 
          <Col md={4}>
            <Label className="form-group has-float-label"> 
            <Input onChange={this.handleChange}
                    value={this.state.salary}
                    type="number" name="salary"
                    step="10000"
                    id="EditUser-salary" 
                    disabled={!isEdit}/>
            <span>salary</span>
            </Label>
          </Col>
          <Col md={4}>
            <Label className="form-group has-float-label"> 
            <Input onChange={this.handleChange}
                    value={this.state.equity}
                    type="number" name="equity"
                    step="0.1"
                    id="EditUser-salary" 
                    disabled={!isEdit}/>
            <span>Equity</span>
            </Label>
          </Col>
          <Col md={4}>
            <Label className="form-group has-float-label"> 
            <Input onChange={this.handleChange}
                    value={this.state.bonus}
                    type="number" name="bonus"
                    step="1000"
                    id="EditUser-salary" 
                    disabled={!isEdit}/>
            <span>Bonus</span>
            </Label>
          </Col>
          
        </Row>
      <hr></hr>
        <Col align="left" >
          <Button color="info" size="sm"
            disabled={!isEdit}> 
            Save</Button>
        </Col>
        <br></br>
      </Form>
    </div>
    )
  }
}



export default UserInfoEditForm;
