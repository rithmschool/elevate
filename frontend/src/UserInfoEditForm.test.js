import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import UserInfoEditForm from "./UserInfoEditForm";

it("renders without crashing", function () {
  shallow(<UserInfoEditForm />);
});

it("matches snapshot", function () {
  let wrapper = shallow(<UserInfoEditForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
