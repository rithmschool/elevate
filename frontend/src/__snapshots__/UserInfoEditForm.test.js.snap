// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot after loading 1`] = `
<div
  className="EditPUserForm container border rounded"
  style={
    Object {
      "backgroundColor": "#F4F6F8",
    }
  }
>
  <Row
    tag="div"
  >
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <h3
        className="text-info"
      >
        User information
      </h3>
    </Col>
    <Col
      align="right"
      md={6}
      sm={
        Object {
          "offset": 0,
          "size": 8,
        }
      }
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <i
        className="fas fa-edit text-info"
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "1.1em",
            "marginTop": "1em",
          }
        }
      />
    </Col>
  </Row>
  <hr />
  <Form
    onSubmit={[Function]}
    tag="form"
  >
    <h5>
      Basic info
    </h5>
    <br />
    <Row
      form={true}
      tag="div"
    >
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-first_name"
            name="first_name"
            onChange={[Function]}
            type="text"
          />
          <span>
            First name
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-last_name"
            name="last_name"
            onChange={[Function]}
            type="text"
          />
          <span>
            Last name
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-email"
            name="email"
            onChange={[Function]}
            type="email"
          />
          <span>
            Email
          </span>
        </Label>
      </Col>
    </Row>
    <hr />
    <h5>
      Company info
    </h5>
    <br />
    <Row
      form={true}
      tag="div"
    >
      <Col
        md={6}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-current_company"
            name="current_company"
            onChange={[Function]}
            type="text"
            value=""
          />
          <span>
            Current company
          </span>
        </Label>
      </Col>
      <Col
        md={6}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-hire_date"
            name="hire_date"
            onChange={[Function]}
            type="date"
            value="2019-09-03"
          />
          <span>
            Hire date
          </span>
        </Label>
      </Col>
    </Row>
    <Row
      form={true}
      tag="div"
    >
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-salary"
            name="salary"
            onChange={[Function]}
            step="10000"
            type="number"
          />
          <span>
            salary
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-salary"
            name="equity"
            onChange={[Function]}
            step="0.1"
            type="number"
          />
          <span>
            Equity
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-salary"
            name="bonus"
            onChange={[Function]}
            step="1000"
            type="number"
          />
          <span>
            Bonus
          </span>
        </Label>
      </Col>
    </Row>
    <hr />
    <Col
      align="left"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Button
        color="info"
        disabled={true}
        size="sm"
        tag="button"
      >
        Save
      </Button>
    </Col>
    <br />
  </Form>
</div>
`;

exports[`matches snapshot while loading 1`] = `
<div
  className="EditPUserForm container border rounded"
  style={
    Object {
      "backgroundColor": "#F4F6F8",
    }
  }
>
  <Row
    tag="div"
  >
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <h3
        className="text-info"
      >
        User information
      </h3>
    </Col>
    <Col
      align="right"
      md={6}
      sm={
        Object {
          "offset": 0,
          "size": 8,
        }
      }
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <i
        className="fas fa-edit text-info"
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "1.1em",
            "marginTop": "1em",
          }
        }
      />
    </Col>
  </Row>
  <hr />
  <Form
    onSubmit={[Function]}
    tag="form"
  >
    <h5>
      Basic info
    </h5>
    <br />
    <Row
      form={true}
      tag="div"
    >
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-first_name"
            name="first_name"
            onChange={[Function]}
            type="text"
          />
          <span>
            First name
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-last_name"
            name="last_name"
            onChange={[Function]}
            type="text"
          />
          <span>
            Last name
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-email"
            name="email"
            onChange={[Function]}
            type="email"
          />
          <span>
            Email
          </span>
        </Label>
      </Col>
    </Row>
    <hr />
    <h5>
      Company info
    </h5>
    <br />
    <Row
      form={true}
      tag="div"
    >
      <Col
        md={6}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-current_company"
            name="current_company"
            onChange={[Function]}
            type="text"
            value=""
          />
          <span>
            Current company
          </span>
        </Label>
      </Col>
      <Col
        md={6}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-hire_date"
            name="hire_date"
            onChange={[Function]}
            type="date"
            value="2019-09-03"
          />
          <span>
            Hire date
          </span>
        </Label>
      </Col>
    </Row>
    <Row
      form={true}
      tag="div"
    >
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-salary"
            name="salary"
            onChange={[Function]}
            step="10000"
            type="number"
          />
          <span>
            salary
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-salary"
            name="equity"
            onChange={[Function]}
            step="0.1"
            type="number"
          />
          <span>
            Equity
          </span>
        </Label>
      </Col>
      <Col
        md={4}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          className="form-group has-float-label"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            disabled={true}
            id="EditUser-salary"
            name="bonus"
            onChange={[Function]}
            step="1000"
            type="number"
          />
          <span>
            Bonus
          </span>
        </Label>
      </Col>
    </Row>
    <hr />
    <Col
      align="left"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Button
        color="info"
        disabled={true}
        size="sm"
        tag="button"
      >
        Save
      </Button>
    </Col>
    <br />
  </Form>
</div>
`;
